main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TranslationFieldAndReferenceValidator.java

```diff
@@ -189,6 +200,52 @@ private static boolean isMissingOrUnexpectedField(
     return true;
   }
 
+  private Optional<?> resolveRecord(
```

Comment:
So this gets into the extended discussion we had about key ordering in https://bit.ly/gtfs-validator-multi-column-primary-keys.  Namely, translations.txt effectively defines a primary key order to be used when performing entity resolution from the recordId + recordSubId fields.  Indeed, the first version of this pull request has a more generic method here, but it hinged on determining the implicit key ordering from the order of fields in the Java schema file.  Ultimately, I was worried that relying on that was shaky at best and since the concept of key ordering was translations.txt-specific, I made the call to move the code into the translations.txt-specific validator.

Now, I will definitely concede that the resulting code, while type-safe, is pretty ugly :)  It's kind of begging for code-generation, but I wasn't sure I wanted to apply that hammer again quite yet.

Namely, I think the arguments about key order are still there.  But maybe there is a middle path here?

Maybe we could define `enum TranslationRecordIdType { RECORD_ID, RECORD_SUB_ID }` and then add a field to PrimaryKey:

```
@interface PrimaryKey {
  ...
  TranslationRecordIdType translationRecordIdType() default RECORD_ID;
}
```

Basically, instead of relying on key-order, we are directly tying the @PrimaryKey definition to the semantics of `translations.txt`.  And because of the default value semantics, we'd really only need to define RECORD_SUB_ID overrides for a handful of fields on existing multi-column schemas.  Generating an appropriate generic lookup method from those annotations should be doable and it doesn't lock us into a more general argument about primary key ordering.

Anyone have thoughts?  @aababilov or @maximearmstrong

Reply:
I would agree with that last idea - `enum TranslationRecordIdType` could be a good middle path. Do you think it should in this PR or another?

I kinda agree with both of you - a more generic validator would be good, but the solution brought by @bdferris-v2 makes sense for what we are trying to accomplish here. I would be okay with this being the initial solution and then we can consider a refactor using `enum TranslationRecordIdType`. Do you think it'd be a good plan?

Reply:
I'd be willing to update this PR if you both thought this was a path worth taking (aka the `TranslationRecordIdType` proposal).

Reply:
I'm fine with that. What about you @aababilov?

Reply:
I love this idea! Let's add TranslationRecordIdType in this PR.

Reply:
Done